safe-modules/liveness-module-v2.md

```diff
@@ -0,0 +1,139 @@
+to sign (e.g., keys are lost) or if signers are malicious and refuse to sign. We must address both
+problems if we want to avoid multisig liveness failures. The existing `LivenessGuard` is
+insufficient in this regard because it does not handle malicious signers.
+
+Any system to address the above must also scale sufficiently such that a signer is not
+unnecessarily burdened by the liveness mechanism (this is a product constraint). The
+`LivenessGuard` is also insufficient here because it does not scale across multiple chains and it
+demands signatures on a regular basis (as opposed to only demanding signatures as needed).
+
+## Customer Description
+
+<!-- Provide a brief summary of the customers for this design document. -->
+
+The customers for this design doc are any participants in the multisig accounts that would utilize
+this module, as well as any 3rd parties who rely on the security properties of these accounts.
+
+### Customer Reviewers
+
+<!-- Identify at least one customer who should be involved in the review of this document. -->
+
+- Coinbase (potentially impacted 3rd party)
+- Uniswap (`LivenessModule` user)
+- Optimism Foundation (`LivenessModule` user)
+
+## Requirements and Constraints
+
+<!-- Identify the solution requirements and any additional design constraints from the Context and
+Problem Statement section in a bulleted list. -->
+
+- Solve both the owner liveness and account liveness problems *securely*.
+- Keep code as minimal as possible to avoid complex logic in Safe modules.
+- Reduce mental overhead for Safe owners and multisig leads.
+
+## Proposed Solution
+
+<!-- Explain the solution that you believe best addresses the problem described above. -->
+
+We propose the introduction of a new module, `LivenessModuleV2` (name is unimportant) to replace
+the existing `LivenessGuard`/`LivenessModule`. `LivenessModuleV2` would employ a challenge/response
+mechanism as an alternative to the existing poke mechanism inside of `LivenessGuard`.
+
+### Singleton
+
+`LivenessModuleV2` is a singleton that exists at a known address on all chains and has no
+constructor parameters. Any Safe on any network can choose to configure the module and then enable
+the module within the Safe.
+
+### Configuration
+
+A Safe configures the module by setting a challenge period, a challenge bond, and a fallback owner.
+
+### Challenge Protocol
+
+Any user may trigger an Account Liveness Challenge by targeting a specific Safe address. In
+response to this challenge, a threshold of owners (same threshold as the Safe itself) must sign a
+message that confirms the liveness of the safe (e.g., a message over the Safe address and the
+current timestamp). If the Safe fails to reach this threshold, all owners are removed from the
+account and ownership is transferred to the fallback account. If the Safe successfully reaches the
+threshold, the challenge fails and the Safe receives the bond as a reward.
```

Comment:
I would simplify even further and instead of talking about a bond, I would consider it as a payment. The challenger always has to pay the amount, and it always be credited to the Safe.

In all likelihood, the challenge would be done by someone that trusts the fallback owner, which then can arrange for the payment to be reimbursed if necessary. Not implementing a path in the module where assets leave the Safe is much safer.

Reply:
Hm, interesting. I like it. Note that the funds wouldn't leave the module until a challenge is successful, but payment is fine, it works fine for the cases we are expecting.

Reply:
I like payment as well - removes a lot of math and tracking.